tcms/rpc/tests/test_testcase.py

```diff
@@ -173,12 +174,39 @@ def test_filter_query_none(self):
         self.assertIn("author", result[0])
         self.assertIn("default_tester", result[0])
         self.assertIn("reviewer", result[0])
+        self.assertIn("setup_duration_sec", result[0])
+        self.assertIn("testing_duration_sec", result[0])
+        self.assertIn("expected_duration_sec", result[0])
 
     def test_filter_by_product_id(self):
         cases = self.rpc_client.TestCase.filter({"category__product": self.product.pk})
         self.assertIsNotNone(cases)
         self.assertEqual(len(cases), self.cases_count)
 
+    @parameterized.expand(
+        [
+            ("both_values_are_not_set", {}, 0, 0, 0),
+            ("setup_duration_is_not_set", {"testing_duration": timedelta(minutes=5)}, 0, 300, 300),
+            ("testing_duration_is_not_set", {"setup_duration": timedelta(seconds=45)}, 45, 0, 45),
+            (
+                "both_values_are_set",
+                {"setup_duration": timedelta(seconds=45), "testing_duration": timedelta(minutes=5)}, 45,
+                300,
+                345
+            )
+        ]
+    )
+    def test_duration_properties_in_result(
+        self, _name, init_dict, setup_duration_sec, testing_duration_sec, expected_duration_sec
+        ):
+        testcase = TestCaseFactory(**init_dict)
+        result = self.rpc_client.TestCase.filter({"pk": testcase.pk})
+
+        self.assertIsNotNone(result)
+        self.assertAlmostEqual(result[0]["setup_duration_sec"], setup_duration_sec)
+        self.assertAlmostEqual(result[0]["testing_duration_sec"], testing_duration_sec)
+        self.assertAlmostEqual(result[0]["expected_duration_sec"], expected_duration_sec)
```

Comment:
Aren't these exactly equal? Why use assertAlmostEqual ?

Reply:
Maybe not in this case.